pkg/cmd/server/admission/init.go

```diff
@@ -22,6 +23,7 @@ type PluginInitializer struct {
 	RESTClientConfig      restclient.Config
 	Informers             shared.InformerFactory
 	ClusterQuotaMapper    clusterquotamapping.ClusterQuotaMapper
+	DefaultRegistryFn     imageapi.DefaultRegistryFunc
```

Comment:
It's already a significant limitation - none of the image stream code looks
at that info, so anyone making different decisions an the kubelet level is
breaking the platform.

On Tue, Aug 9, 2016 at 9:02 AM, David Eads <notifications@github.com> wrote:

> In pkg/cmd/server/admission/init.go
> <https://github.com/openshift/origin/pull/8995#discussion_r74052868>:
>
> > @@ -22,6 +23,7 @@ type PluginInitializer struct {
> >  	RESTClientConfig      restclient.Config
> >  	Informers             shared.InformerFactory
> >  	ClusterQuotaMapper    clusterquotamapping.ClusterQuotaMapper
> > +	DefaultRegistryFn     imageapi.DefaultRegistryFunc
>
> Master config should eventually allow alternate names for the registry,
> and it'll be an array of strings. Kubelets will not be allowed to have
> different decisions for what an image registry is.
>
> And you're saying, "not in this pull"? I can accept hat, but it's a
> significant limitation.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/8995/files/f96f5e56ba70c7c0f24f875621efdf1ef45ce3c0#r74052868>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABG_pybKbwfCjY4XvWQvvAC_iSjfhlTKks5qeHp6gaJpZM4IlDlb>
> .
>